src/main/java/io/cryostat/recordings/LongRunningRequestGenerator.java

```diff
@@ -0,0 +1,268 @@
+    public static final String ACTIVE_REPORT_ADDRESS =
+            "io.cryostat.recordings.ArchiveRequestGenerator.ActiveReportRequest";
+    private static final String ARCHIVE_RECORDING_SUCCESS = "ArchiveRecordingSuccess";
+    private static final String ARCHIVE_RECORDING_FAIL = "ArchiveRecordingFailed";
+    private static final String GRAFANA_UPLOAD_SUCCESS = "GrafanaUploadSuccess";
+    private static final String GRAFANA_UPLOAD_FAIL = "GrafanaUploadFailed";
+    private static final String REPORT_SUCCESS = "ReportSuccess";
+    private static final String REPORT_FAILURE = "ReportFailure";
+    private final ExecutorService executor;
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Inject private EventBus bus;
+    @Inject private RecordingHelper recordingHelper;
+    @Inject ReportsService reportsService;
+
+    @ConfigProperty(name = ConfigProperties.CONNECTIONS_FAILED_TIMEOUT)
+    Duration timeout;
+
+    public LongRunningRequestGenerator(ExecutorService executor) {
+        this.executor = executor;
+    }
+
+    public Future<String> performArchive(ArchiveRequest request) {
+        Objects.requireNonNull(request.recording);
+        return executor.submit(
+                () -> {
+                    logger.info("Job ID: " + request.getId() + " submitted.");
+                    try {
+                        String rec =
+                                recordingHelper
+                                        .archiveRecording(request.recording, null, null)
+                                        .name();
+                        logger.info("Recording archived, firing notification");
+                        bus.publish(
+                                MessagingServer.class.getName(),
+                                new Notification(
+                                        ARCHIVE_RECORDING_SUCCESS,
+                                        Map.of("jobId", request.getId(), "recording", rec)));
+                        return request.getId();
+                    } catch (Exception e) {
+                        logger.info("Archiving failed");
+                        bus.publish(
+                                MessagingServer.class.getName(),
+                                new Notification(
+                                        ARCHIVE_RECORDING_FAIL, Map.of("jobId", request.getId())));
+                        throw new CompletionException(e);
+                    }
+                });
+    }
+
+    @ConsumeEvent(value = ARCHIVE_ADDRESS)
+    public void onMessage(ArchiveRequest request) {
+        try {
+            performArchive(request);
+        } catch (Exception e) {
+            logger.warn("Exception thrown while servicing request: ", e);
+        }
+    }
+
+    @ConsumeEvent(value = GRAFANA_ARCHIVE_ADDRESS, blocking = true)
+    public void onMessage(GrafanaArchiveUploadRequest request) {
+        try {
+            logger.info("Job ID: " + request.getId() + " submitted.");
```

Comment:
All these log messages in this class, for example, should be TRACE or DEBUG level. Except the ones in `catch` clauses, then WARN or ERROR makes sense.